great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -405,19 +413,22 @@ def test_connection(self, test_assets: bool = True) -> None:
             )
         if self.assets and test_assets:
             for asset in self.assets.values():
+                asset._datasource = self
```

Comment:
When will these be not set? It seems strange to have to reach into the asset to set this.

Reply:
This isn't set because in this case we are passing `DataAsset`s into the `Datasource` constructor. You can't test the `DataAsset` without a `SQLDatasource` `_engine`.